autoload/lsp/util.vim

```diff
@@ -34,7 +34,13 @@ enddef
 
 # Empty out the LSP server trace logs
 export def ClearTraceLogs(fname: string)
-  writefile([], fname)
+  var fullname = $'{lsp_log_dir}{fname}'
+  # If file existed but not writable
+  if filewritable(fullname) != 1 && filereadable(fullname)
```

Comment:
If the log and error files are not writable, then TraceLog() function also will fail.  So checking whether the file is writable or not here is not sufficient.

The bug here is that the directory name is not prepended to the file name.  Can you change this function to:
```
    writefile([], $'{lsp_log_dir}{fname}'
```

Reply:
yes, the main point was it should be `fullname` (vs `fname`), 
// but before tracelog it would be clearlog first, so did a writable check (into to msg) was a hint, no harm.